NetworkPkg/HttpBootDxe/HttpBootClient.c

```diff
@@ -1097,6 +1129,58 @@ HttpBootGetBootFile (
     }
   }
 
+  //
+  // Add HTTP header field 5 (optional): Range
+  //
+  if (ResumingOperation) {
+    // Resuming a failed download. Prepare the HTTP Range Header
+    Status = AsciiSPrint (
+               RangeValue,
+               sizeof (RangeValue),
+               "bytes=%lu-%lu",
+               Private->PartialTransferredSize,
+               Private->BootFileSize - 1
+               );
+    if (EFI_ERROR (Status)) {
+      goto ERROR_3;
+    }
+
+    Status = HttpIoSetHeader (HttpIoHeader, "Range", RangeValue);
+    if (EFI_ERROR (Status)) {
+      goto ERROR_3;
+    }
+
+    DEBUG (
+      (DEBUG_WARN | DEBUG_INFO,
+       "HttpBootGetBootFile: Resuming failed download. Range: %a\n",
+       RangeValue)
+      );
+
+    // Add If-Unmodified-Since header with the value got from the first request when resuming a download!

```

Comment:
@SaloniKasbekar 
I will also create definitions for the headers I'm using that are not defined at Http11.h:

///
/// Last-Modified Response Header
/// The Last-Modified response HTTP header contains a date and time when 
/// the origin server believes the resource was last modified. It is used 
/// as a validator to determine if the resource is the same as the 
/// previously stored one. Less accurate than an ETag header,
// it is a fallback mechanism. Conditional requests containing 
// If-Modified-Since or If-Unmodified-Since headers make use of this field.
///
#define HTTP_HEADER_LAST_MODIFIED "Last-Modified"

///
/// If Unmodified Since Request Header
/// Makes the request for the resource conditional: the server will send
/// the requested resource or accept it in the case of a POST or another
/// non-safe method only if the resource has not been modified after the 
/// date specified by this HTTP header. If the resource has been modified 
/// after the specified date, the response will be a 412 Precondition Failed error.
///
#define HTTP_HEADER_IF_UNMODIFIED_SINCE "If-Unmodified-Since"

What do you think?

Reply:
Yes, it would be good to add those changes in MdePkg. It can be a part of the same PR, but will need to be in a different commit.